### DISCUSSION TITLE
Specifying Range For HttpRangeReader / Writer?

### DISCUSSION THREAD

**Discussion by cmcnicholas** [gildas-lormeau/zip.js#536] [Q&A] [answered]
Library is exactly what I'm looking for and have a PoC going which downloads multiple files with `HttpRangeReader` and saves to disk using `ZipWriter` and passthrough enabled.

When I look at the network traffic for this I can see many sequential fetch requests for `512kb`, is there any way to control the range size requested? I am working with files that are 2-10GB on average and the setup/teardown of 4000+ fetch requests adds significant time vs a single request.

I'm assuming zip writer will be able to yield better with smaller requests than one fetch response for 2+ GB but I would like to strike a balance of maybe 2-10MB per request to keep things moving quickly.

**Comment by gildas-lormeau** [comment, marked as answer]
You can call [`zip.configure({ chunkSize: <desired size in bytes> })`](https://gildas-lormeau.github.io/zip.js/api/interfaces/Configuration.html#chunkSize) to set the chunk size.
